setup/setup.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+set -e
+
+if ! command -v go >/dev/null 2>&1
+then
+    echo "Please install Go."
+    echo "Instructions can be found here: https://golang.org/"
+    exit 1
+fi
+
+if ! command -v git >/dev/null 2>&1
+then
+    echo "Please install Git."
+    echo "Instructions can be found here: "
+    exit 1
+fi
+
+if ! command -v psql >/dev/null 2>&1
+then
+    echo "Please install Postgresql."
+    echo "Instructions can be found here: https://www.postgresql.org/"
+    exit 1
+fi
+
+read -rp "Please enter the database name for archon (default: archondb): " DB_NAME
+if [ ! "$DB_NAME" ]; then
+  DB_NAME="archondb"
+fi
+
+read -rp "Please enter the username for the archon database (default: archonadmin): " ARCHON_USER
+if [ ! "$ARCHON_USER" ]; then
+  ARCHON_USER="archonadmin"
+fi
+
+read -rp "Please enter the password for the archon database (default: psoadminpassword): " ARCHON_PASSWORD
+if [ ! "$ARCHON_PASSWORD" ]; then
+  ARCHON_PASSWORD="psoadminpassword"
+fi
+
+read -rp "Please enter the server address (default: 0.0.0.0/32): " SERVER_IP
+if [ ! "$SERVER_IP" ]; then
+  SERVER_IP="0.0.0.0/32"
+fi
+
+read -rp "Please enter the external server address (default: 127.0.0.1): " EXTERNAL_ADDRESS
+if [ ! "$EXTERNAL_ADDRESS" ]; then
+  SERVER_IP="127.0.0.1"
+fi
+
+# No matter where we're calling this from, we're going to use
+# The root archon directory as our install location.
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Go to the root archon directory.
+pushd "$SCRIPT_DIR"/.. > /dev/null 2>&1
+
+SETUP_DIR="$(pwd)/setup"
+
+echo "Please edit the config file: $SETUP_DIR/config.yaml"
```

Comment:
Not sure I follow why this comment exists. Given this script is (rightly) copying the setup directory and editing the configs for which user input was collected, maybe this should be replaced by a message at the end saying to confirm settings before starting the server or something like that